datasource/src/main/scala/quasar/physical/mongo/Tunnel.scala

```diff
@@ -126,12 +126,12 @@ object Tunnel {
       config: TunnelConfig,
       settings: ClusterSettings,
       blockingPool: BlockingContext)
-      : F[Disposable[F, ClusterSettings]] = for {
+      : F[Disposable[F, ClusterSettings]] = ContextShift[F].evalOn(blockingPool.unwrap) { for {
     jsch <- mkJSch
     sess <- mkSession(jsch, config)
     _ <- setUserInfo(sess, userInfo(config))
     address <- serverAddress(settings)
-    tunnel <- ContextShift[F].evalOn(blockingPool.unwrap)(openTunnel(sess, address))
+    tunnel <- openTunnel(sess, address)
     result <- tunneledSettings(settings, tunnel)
-  } yield Disposable(result, ContextShift[F].evalOn(blockingPool.unwrap)(closeTunnel(sess, tunnel)))
+  } yield Disposable(result, ContextShift[F].evalOn(blockingPool.unwrap)(closeTunnel(sess, tunnel))) }
 }
```

Comment:
Interesting, I would have left the `evalOn` associated with `closeTunnel` outside the scope of the first `evalOn`, so that there wouldn't be any nesting, but I think this is ok.

Reply:
AFAIU we are back in main pool after the `F[Disposable[...` is returned and closing will happen in very different context (unknown, somewhere)

Reply:
Yeah that's why I think it's ok, just had to look twice :)